spyder/plugins/variableexplorer/widgets/collectionseditor.py

```diff
@@ -332,7 +332,10 @@ def data(self, index, role=Qt.DisplayRole):
         if index.column() == 3:
             display = value_to_display(value, minmax=self.minmax)
         else:
-             display = to_text_string(value)
+            if PY2:
+                display = to_text_string(value, encoding="utf-8")

```

Comment:
Tests are failing because we're assuming here that `value` is always a string, but that's wrong, it can be any object.

So please change this line to be

```python
if is_type_text_string(value):
    display = to_text_string(value, encoding="utf-8")
else:
    display = to_text_string(value)
```

(`is_type_text_string` is a function present in `py3compat`).

That should be enough.